components/brave_ads/core/internal/ads/serving/choose/sample_ads.h

```diff
@@ -17,22 +19,28 @@
 namespace brave_ads {
 
 template <typename T>
-double CalculateNormalizingConstant(
+T CalculateNormalizingConstant(const std::vector<T>& scores) {
+  return std::accumulate(scores.cbegin(), scores.cend(), 0.0);
```

Comment:
we suppose that `T` type is double because this uses 0.0 as an initial value. maybe it is better to explicitly specify type `const std::vector<double>& scores`?

Reply:
we could use other types for the input scores, what's the best way to initialize the `std::accumulate` in this case?

Reply:
what types of scores can be used? 
We use only one type for now. Maybe it is better to add generic functions when you have another type? WDYT?

Reply:
right now we use it both with `int` (from `ComputeProbabilities`) and `double` (from `CalculateNormalizingConstantFromPredictors`)

Reply:
so we can use T{} as initializer for std::accumulate